src/Extract.php

```diff
@@ -0,0 +1,159 @@
+<?php
+namespace DrupalComposer\DrupalScaffold;
+
+use Robo\Result;
+use Robo\Task\BaseTask;
+
+/**
+ * Extracts an archive.
+ *
+ * ``` php
+ * <?php
+ * $this->taskExtract($archivePath)
+ *  ->to($destination)
+ *  ->run();
+ * ?>
+ * ```
+ *
+ * @method to(string) location to store extracted files
+ */
+class Extract extends BaseTask
+{
+    use \Robo\Common\DynamicParams;
+    use \Robo\Common\Timer;
+
+    protected $filename;
+    protected $to;
+
+    public function __construct($filename)
+    {
+        $this->filename = $filename;
+    }
+
+  function run() {
+    if (!file_exists($this->filename)) {
+      $this->printTaskError("File {$this->filename} does not exist");
+      return false;
+    }
+    if (!($mimetype = static::archiveType($this->filename))) {
+      $this->printTaskError("Could not determine type of archive for {$this->filename}");
+      return false;
+    }
+
+    // We will first extract to $extractLocation and then move to $this->to
+    $extractLocation = static::getTmpDir();
+    @mkdir($extractLocation);
+    @mkdir(dirname($this->to));
+
+    // Perform the extraction of a zip file.
+    if (($mimetype == 'application/zip') || ($mimetype == 'application/x-zip')) {
+      $command = "unzip {$this->filename} -d --destination=$extractLocation";
+    }
+    // Otherwise we have a possibly-compressed Tar file.
+    // If we are not on Windows, then try to do "tar" in a single operation.
+    else {
+      $tar_compression_flag = '';
+      if ($mimetype == 'application/x-gzip') {
+        $tar_compression_flag = 'z';
+      }
+      elseif ($mimetype == 'application/x-bzip2') {
+        $tar_compression_flag = 'j';
+      }
+      $command = "tar -C $extractLocation -x{$tar_compression_flag}f {$this->filename}";
```

Comment:
I am reluctant to use TarDownloader, because I want to move Extract.php to codegyre/Robo.